indico/modules/events/registration/controllers/management/fields.py

```diff
@@ -49,6 +50,12 @@ def _check_input_type(self, input_type, **kwargs):
         if input_type not in get_field_types():
             raise ValidationError('Invalid field type')
 
+    @validates('retention_period')
+    def _check_retention_period(self, retention_period, **kwargs):
+        field = self.context['field']
+        if retention_period and field.view_data.get('fieldIsRequired'):
```

Comment:
`view_data` is meant only for the front-end; check if the field is a personal data field and has `field.personal_data_type.is_required` instead